<p><strong>Balaji Iyer (PhD ABD Candidate)</strong></p>
<p><strong>Artificial Intelligence | Machine Learning | Data
Science</strong></p>
<p>Cincinnati | 513-641-7176 |<a
href="https://www.iyerbalaji.com">https://www.iyerbalaji.com</a>|<a
href="mailto:iyerbs@mail.uc.edu">iyerbs@mail.uc.edu</a> | <a
href="https://www.linkedin.com/in/balajisiyer1485">LinkedIn</a> | <a
href="https://github.com/dlab-arp">GitHub</a> |</p>
<p><strong>Last Updated on 25-March-2025</strong></p>
<p><strong>Summary</strong></p>
<p>Data scientist and software developer with 10+ years of experience in
applied DL, ML, and statistical modeling. As a founding research student
of the lab, pioneered projects across multiple domains of biology and
medicine ranging from genomics to neuroscience. Contributed towards 12
projects, resulting in 12 technical manuscripts and 3 talks. Nurtured
next generation researchers by mentoring 2 masters, 4 undergraduates,
and a high school student across five national and three international
collaborations.</p>
<p>Leveraged unsupervised and full/self/semi-supervised learning
paradigms to develop generative and discriminative models encompassing a
diverse array of data modalities ranging from 1D to 4D. Proficient in
addressing challenges such as sparse annotations, data imbalance, and
poor data quality, regardless of dataset size. Extensive expertise in
established deep learning paradigms, including transformers and
diffusion models, while continuously striving to stay abreast of the
latest advancements such as Mamba and KAN.</p>
<p><strong>Research Experience</strong></p>
<p><strong>PhD Candidate, Prasath Lab</strong></p>
<p><strong>Dept. of Computer Science, University of Cincinnati &amp;
Cincinnati Children’s Hospital Medical Center 2019 –
Present</strong></p>
<ul>
<li><p><strong>Reimagining Interpretability in Segmentation: A
Multi-Scale Coherence Framework</strong></p>
<ul>
<li><p>Pioneered an entirely new perspective of Interpretability for
Segmentation through XOR-based Deep Supervision—lighting up hidden
multi-scale cues in real time and unveiling how the model forms
decisions.</p></li>
<li><p>Introduced novel consistency heatmaps, exposing exactly where and
why the network struggles, letting clinicians “see” AI uncertainty like
never before.</p></li>
<li><p>Created an innovative Out-Of-Distribution detector through
game-changing metrics (Transition Dice &amp; Outlier Consistency) that
proactively flags distributional shifts—setting a new bar for
trustworthy medical image segmentation.</p></li>
</ul></li>
<li><p><strong>maxATAC - Maximizing insights from ATAC-seq. (Assay for
Transposase-Accessible Chromatin) data</strong></p>
<ul>
<li><p>Led the development of a dilated 1D CNN model and specialized
training strategy, resulting in maxATAC, a publicly available collection
of 127 Transcription Factor (TF) -specific CNN models capable of
predicting TF binding sites from ATAC-seq data, that generalizes well
even on Out Of Distribution cell lines.</p></li>
<li><p>Currently developing maxATAC v2 with Transformer architecture for
enhanced performance and interpretability.</p></li>
</ul></li>
<li><p><strong>DeepImmuno: deep learning-empowered prediction and
generation of immunogenic peptides for T-cell immunity</strong></p>
<ul>
<li><p>Developed a deep learning-based model,
<strong>DeepImmuno-CNN</strong>, for the prediction of immunogenic
peptides from MHC-peptide pairs, using a beta-binomial distribution for
continuous immunogenic score calculation.</p></li>
<li><p>The model outperforms two other prominent algorithms (DeepHLApan,
IEDB), with better precision and recall across challenging datasets like
cancer neoantigen and SARS-CoV-2.</p></li>
<li><p>Created <strong>DeepImmuno-GAN</strong>, for generating synthetic
immunogenic peptides with real antigen-like properties, enabling the
generation of high-quality synthetic datasets for training and
biological applications.</p></li>
<li><p><strong>DeepImmuno</strong> is available as open-source code on
GitHub and as a user-friendly web interface portal for immunogenicity
prediction. Currently, mentoring a Masters student for DeepImmuno
v2.</p></li>
</ul></li>
<li><p><strong>AI‑Driven Gait Classification from Low Resolution Videos
for Cerebral Palsy (CP) Patients</strong></p></li>
</ul>
<ul>
<li><p>Developed an AI-driven video analysis model to streamline gait
analysis for CP patients, reducing resource requirements and test time
by 95%, aiming for a portable app-based monitoring system.</p></li>
<li><p>Created an autoencoder-based CNN model using self-supervised
learning, designed data collection protocols, and developed data
pipelines.</p></li>
<li><p>Achieved 65% OGS accuracy with a small dataset, indicating
potential performance improvements with more data.</p></li>
<li><p>Mentored an international master's candidate on VAE and Memory
Augmented Autoencoder models, resulting in an IEEE publication.</p></li>
</ul>
<ul>
<li><p><strong>Benchmarking HEp-2 (Human Epithelial) Cell Segmentation
Methods in Indirect Immunofluorescence (IIF) Images - Standard Models to
Deep Learning</strong></p>
<ul>
<li><p>Benchmarked 20 state-of-the-art DL segmentation models, including
CNN and Transformer architectures, and proposed GAN-based generative
models. Developed a novel domain specific pretraining strategy and
proposed 2 different strategies for data augmentation and
comprehensively evaluated their performance.</p></li>
<li><p>Achieved a State-of-the-Art Dice score of 92.2% with results are
under peer review.</p></li>
</ul></li>
<li><p><strong>NeuroGleam: Illuminating Small Vessel Disease (SVD)
Detection through Deep Learning based Segmentation of Brain MRI White
Matter Hyperintensities (WMH)</strong></p>
<ul>
<li><p>Developed a DL-based WMH segmentation model for improved SVD
detection from brain MRI scans, incorporating a two-stage NN model with
CNN and Vision Transformer architecture and tested extensively on public
(MICCAI WMH Challenge) and in-house clinical dataset</p></li>
<li><p>Achieved a high Dice score of 80% on the public dataset and
established a new State-of-the-Art score of 75% on the clinical in-house
dataset compared to contemporary models.</p></li>
</ul></li>
</ul>
<p><strong>Skills</strong></p>
<ul>
<li><p><strong>Programming:</strong> Python | MATLAB| Julia | C | C++ |
Arduino C</p></li>
<li><p><strong>Libraries:</strong> Numpy | Scipy | Pandas | Tensorflow |
Keras | PyTorch | Docker</p></li>
<li><p><strong>Modeling:</strong> CNN | Transformer | RNN | LSTM | VAE |
GAN | Diffusion | Interpretability | RAG| Unsupervised | Self-Supervised
| Semi-Supervised | Mamba | KAN</p></li>
</ul>
<p><strong>Education</strong></p>
<p><strong>PhD, Computer Science, University of Cincinnati Expected
Summer 2025</strong></p>
<p><strong>Dissertation</strong> – Multimodal Artificial Intelligence
for Bioinformatics Data</p>
<p><strong>MS, Computer Science, University of Cincinnati Dec
2018</strong></p>
<p><strong>Thesis</strong> – Spindle bearing health prognosis using
Machine Learning and Signal Processing.</p>
<p><strong>BS, Instrumentation &amp; Control Engineering, University of
Pune, India Jun 2008</strong></p>
<p><strong>Research &amp; Teaching Assistantships</strong></p>
<p><strong>Graduate Research Assistantship</strong></p>
<ul>
<li><p>Assisted with 3 NIH grant proposals, contributed to 12
collaborative interdisciplinary projects in the biology and medical
domain ranging from genomics, immunology, neuroscience, radiomics and
gait analysis. Represented the lab in 6 conferences with 3 talks and 3
posters.</p></li>
</ul>
<p><strong>Graduate Teaching Assistantship</strong></p>
<ul>
<li><p>Courses: Intelligent Data Analysis | C++ Programming |
Introduction to Computer Systems</p></li>
</ul>
<p><strong>Publications</strong></p>
<ul>
<li><p><strong><u>Balaji Iyer,</u></strong> et.al. " Reimagining
Interpretability in Segmentation: A Multi-Scale Coherence Framework"
(under review)</p></li>
<li><p>T. Cazares,…, <strong><u>Balaji Iyer,</u></strong> et.al.
"maxATAC: Genome-scale transcription-factor binding prediction from
ATAC-seq with deep neural networks." PLOS Computational
Biology.</p></li>
<li><p>G. Li, <strong><u>Balaji Iyer,</u></strong> et.al. "DeepImmuno:
deep learning-empowered prediction and generation of immunogenic
peptides for T-cell immunity." Briefings in bioinformatics.</p></li>
<li><p>A. Gundawar,…, <strong><u>Balaji Iyer</u></strong>, et.al. "On
the Performance of new Higher Order Transformation Functions for Highly
Efficient Dense Layers." Neural Processing Letters.</p></li>
<li><p>N. Gaddis,..., <strong><u>Balaji Iyer</u></strong>, et.al.
LungMAP portal ecosystem: Systems-level exploration of the lung.
American Journal of Respiratory Cell and Molecular Biology.</p></li>
<li><p>G. Sumi, <strong><u>Balaji Iyer</u></strong>, et.al. Gait
analysis for cerebral palsy using memory-augmented auto-encoder model.
11th International Conference on Informatics, Electronics &amp; Vision
(ICIEV), Proc. IEEE. (<strong>Excellent Paper Award</strong>)</p></li>
<li><p><strong><u>Balaji Iyer</u></strong>, et.al. Benchmarking HEp‑2
cell segmentation methods in indirect immunofluorescence images. (under
review)</p></li>
<li><p><strong><u>Balaji Iyer</u></strong>, et.al. AI-driven gait
parameters estimation from videos for cerebral palsy patients (In
preparation)</p></li>
<li><p><strong><u>Balaji Iyer</u></strong>, et.al. NeuroGleam:
Illuminating Small Vessel Disease Detection through Deep Learning based
Segmentation of Brain MRI White Matter Hyperintensities. (In
preparation)</p></li>
<li><p><strong><u>Balaji Iyer</u></strong>, et.al. ConIFSeg - Confocal
immunofluorescence image segmentation with deep learning for discerning
developing lung structures. (In preparation)</p></li>
<li><p>Smruti Deoghare,…, <strong><u>Balaji Iyer</u></strong>, et.al.
PICCLineNet: Detecting peripherally inserted central catheter (PICC)
lines and tip malpositioning in neonate X-ray images using artificial
intelligence (AI) models (In Preparation)</p></li>
<li><p><strong><u>Balaji Iyer</u></strong>, et.al. Predicting ICH
patient outcome from brain CT scans using an ensemble deep learning
framework (In Preparation)</p></li>
<li><p>Prasath Lab, Book chapter: “Application of Multimodal Deep
Learning in Medicine” (under review).</p></li>
</ul>
<p><strong>Presentations</strong></p>
<ul>
<li><p><strong><u>Balaji Iyer</u></strong>, et.al. Deep learning-based
segmentation of human epithelial type-2 (HEp-2) cells using indirect
immunofluorescence (IIF) images. Advancing Healthcare Innovation Summit
(AHIS), November 2023. Talk</p></li>
<li><p><strong><u>Balaji Iyer</u></strong>, et.al. AI-driven gait
parameters estimation from videos for cerebral palsy patients. Advancing
Healthcare Innovation Summit (AHIS), November 2023. Talk.
(<strong>Aspiring Innovator Award</strong>, <strong>2nd Best
Presentation Award</strong>)</p></li>
<li><p><strong><u>Balaji Iyer</u></strong>, et.al. Predicting ICH
patient outcome from brain CT scans using an ensemble deep learning
framework. Advanced Computational Neuroscience Network (ACNN),
University of Michigan, Ann Arbor, MI, USA. September 2019.
Talk</p></li>
<li><p><strong><u>Balaji Iyer</u></strong>, et.al. ConIFSeg - Confocal
immunofluorescence image segmentation with deep learning for discerning
developing lung structures. First Annual Cincinnati Children's Research
Symposium, November 2023. Poster presentation.</p></li>
<li><p><strong><u>Balaji Iyer</u></strong>, et.al. Deep learning based
segmentation of human epithelial type-2 (HEp-2) cells using indirect
immunofluorescence (IIF) images. First Annual Cincinnati Children's
Research Symposium, November 2023. Poster presentation.</p></li>
<li><p>Smruti Deoghare,…, <strong><u>Balaji Iyer</u></strong>, et.al.
PICCLineNet: Detecting peripherally inserted central catheter (PICC)
lines and tip malpositioning in neonate X-ray images using artificial
intelligence (AI) models. First Annual Cincinnati Children's Research
Symposium, May 2023. Poster Presentation</p></li>
<li><p>Samuel Hacker, <strong><u>Balaji Iyer</u></strong>, et. al.
Automated ICH Outcome Prediction from CT scans by Ensemble Convolutional
Neural Network Architecture. Capstone Poster Symposium, Aug 2019. Poster
Presentation (<strong>2<sup>nd</sup> Best Presentation
Award</strong>)</p></li>
</ul>
<p><strong>Peer Review</strong></p>
<ul>
<li><p>Journal Reviewer for NCAA. Springer Nature Neural Computing and
Applications</p></li>
<li><p>Journal Reviewer for EJRAI. Elsevier European Journal of
Radiology Artificial Intelligence</p></li>
<li><p>Journal Reviewer for ESA. Elsevier Expert Systems With
Applications</p></li>
<li><p>Journal Reviewer for BSPC. Elsevier Biomedical Signal Processing
and Control</p></li>
</ul>
